src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -120,4 +120,26 @@ export function findOccurencesOf(document: vscode.TextDocument, keyword: string)
     return ranges
 }
 
+const HIGHLIGHTER = vscode.window.createTextEditorDecorationType({
+    backgroundColor: new vscode.ThemeColor('list.focusHighlightForeground'),
+})
+
+export async function highlightDocument(registry: LogStreamRegistry, uri: vscode.Uri): Promise<void> {
+    const textEditor = registry.getTextEditor(uri)
+    const logData = registry.getLogData(uri)
+
+    if (!logData) {
+        throw new Error(`Missing log data in registry for uri key: ${uriToKey(uri)}. Unable to highlight`)
+    }
+
+    if (!textEditor) {
+        throw new Error(`Missing textEditor in registry for uri key: ${uriToKey(uri)}. Unable to highlight`)
+    }
+
+    if (logData.parameters.filterPattern) {
+        const ranges = findOccurencesOf(textEditor.document, logData.parameters.filterPattern)
```

Comment:
doesn't look like `findOccurencesOf()` is specific to CWL, so it should live in `src/shared/utilities/textDocumentUtilities.ts`